sherpa/astro/io/pyfits_backend.py

```diff
@@ -163,62 +206,67 @@ def _try_vec(hdu, name, size=2, dtype=SherpaFloat, fix_type=False):
         col = col.astype(dtype)
 
     if col is None:
-        return numpy.array([None]*size)
+        return numpy.array([None] * size)
 
     return col
 
+
 def _require_col(hdu, name, dtype=SherpaFloat, fix_type=False):
     col = _try_col(hdu, name, dtype, fix_type)
     if col is None:
         raise IOErr('reqcol', name, hdu._file.name)
     return col
 
+
 def _require_tbl_col(hdu, name, dtype=SherpaFloat, fix_type=False):
     col = _try_tbl_col(hdu, name, dtype, fix_type)
     if len(col) > 0 and col[0] is None:
         raise IOErr('reqcol', name, hdu._file.name)
     return col
 
+
 def _require_vec(hdu, name, size=2, dtype=SherpaFloat, fix_type=False):
     col = _try_vec(hdu, name, size, dtype, fix_type)
     if numpy.equal(col, None).any():
         raise IOErr('reqcol', name, hdu._file.name)
     return col
 
+
 def _try_col_or_key(hdu, name, dtype=SherpaFloat, fix_type=False):
     col = _try_col(hdu, name, dtype, fix_type)
     if col is not None:
         return col
     return _try_key(hdu, name, fix_type, dtype)
 
+
 def _try_vec_or_key(hdu, name, size, dtype=SherpaFloat, fix_type=False):
     col = _try_col(hdu, name, dtype, fix_type)
     if col is not None:
         return col
-    return numpy.array([_try_key(hdu, name, fix_type, dtype)]*size)
+    return numpy.array([_try_key(hdu, name, fix_type, dtype)] * size)
 
 
-
-## Read Functions ##
+# Read Functions
 
 def read_table_blocks(arg, make_copy=False):
 
     filename = ''
     hdus = None
-    if type(arg) is pyfits.HDUList:
+    if type(arg) is fits.HDUList:
```

Comment:
We should use `isinstance` rather than comparing the class. Even better, we should rely on duck typing, although this is probably more complicated at this stage.